### PR TITLE
Support defaults, sequences, constraints for inserts into compressed chunks

### PR DESCRIPTION
**Pull request by gayyappan**

### CODE REVIEW

**Review by svenklemm — APPROVED** [review]

File: src/nodes/chunk_insert_state.c [hunk 1]
```diff
@@ -562,33 +589,61 @@ extern ChunkInsertState *
...
 	MemoryContextSwitchTo(cis_context);
-	resrelinfo = create_chunk_result_relation_info(dispatch, rel);
+	relinfo = create_chunk_result_relation_info(dispatch, rel);
+	if (!is_compressed)
+		resrelinfo = relinfo;
+	else
+	{
+		/* insert the tuple into the compressed chunk instead */
+		resrelinfo = create_compress_chunk_result_relation_info(dispatch, compress_rel);
+	}
```
Comment (mkindahl) [inline src/nodes/chunk_insert_state.c:646]: Suggest to use `{` on both branches or none.

File: tsl/src/compression/compression.c [hunk 2]
```diff
@@ -1514,3 +1518,164 @@ update_compressed_chunk_relstats(Oid uncompressed_relid, Oid compressed_relid)
...
+	List *htcols_list = NIL;
+	int i = 0, cclen;
+	const ColumnCompressionInfo **ccinfo;
+	TupleDesc in_desc = RelationGetDescr(in_rel);
+	TupleDesc out_desc = RelationGetDescr(out_rel);
+	int16 *in_column_offsets;
+	int n_keys;
+	const ColumnCompressionInfo **keys;
+
+	CompressSingleRowState *cr = palloc(sizeof(CompressSingleRowState));
```
Comment (mkindahl) [inline tsl/src/compression/compression.c:1545]:
```suggestion
	CompressSingleRowState *cr = palloc0(sizeof(CompressSingleRowState));
```

File: tsl/src/compression/compression.c [hunk 3]
```diff
@@ -1514,3 +1518,164 @@ update_compressed_chunk_relstats(Oid uncompressed_relid, Oid compressed_relid)
...
+	cr->out_slot =
+		MakeSingleTupleTableSlotCompat(RelationGetDescr(out_rel), table_slot_callbacks(out_rel));
+	cr->in_rel = in_rel;
+	cr->out_rel = out_rel;
+
+	/* get compression properties for hypertable */
+	htcols_list = ts_hypertable_compression_get(srcht_id);
+	cclen = list_length(htcols_list);
+	ccinfo = palloc(sizeof(ColumnCompressionInfo *) * cclen);
```
Comment (mkindahl) [inline tsl/src/compression/compression.c:1554]:
```suggestion
	ccinfo = palloc0(sizeof(ColumnCompressionInfo *) * cclen);
```

File: tsl/src/compression/compression.c [hunk 4]
```diff
@@ -1514,3 +1518,164 @@ update_compressed_chunk_relstats(Oid uncompressed_relid, Oid compressed_relid)
...
+
+static TupleTableSlot *
+compress_singlerow(CompressSingleRowState *cr, TupleTableSlot *in_slot)
+{
+	Datum *invalues, *out_values;
+	bool *out_isnull;
+	TupleTableSlot *out_slot = cr->out_slot;
+	RowCompressor *row_compressor = &cr->row_compressor;
+
+	slot_getallattrs(in_slot);
```
Comment (mkindahl) [inline tsl/src/compression/compression.c:1594]: You have already fetched all attributes in the calling function.

**Review by mkindahl — APPROVED** [review]

File: tsl/src/compression/compression.c (hunk 4 above)
Comment (gayyappan) [reply tsl/src/compression/compression.c]: fixed.

File: src/nodes/chunk_insert_state.c [hunk 5]
```diff
@@ -562,33 +589,61 @@ extern ChunkInsertState *
...
-	Relation rel, parent_rel;
+	Relation rel, parent_rel, compress_rel;
 	MemoryContext old_mcxt;
 	MemoryContext cis_context = AllocSetContextCreate(dispatch->estate->es_query_cxt,
 													  "chunk insert state memory context",
 													  ALLOCSET_DEFAULT_SIZES);
 	OnConflictAction onconflict_action = ts_chunk_dispatch_get_on_conflict_action(dispatch);
-	ResultRelInfo *resrelinfo;
+	ResultRelInfo *resrelinfo, *relinfo;
+	bool is_compressed = (chunk->fd.compressed_chunk_id != 0);
```
[the remark this answers is not in the document]
Comment (gayyappan) [reply src/nodes/chunk_insert_state.c]: yes, expect some macros to be added when we use the newly added status flag.

**Pull request merged**